boa_engine/src/value/operations.rs

```diff
@@ -195,22 +195,41 @@ impl JsValue {
     }
 
     /// Perform the binary `**` operator on the value and return the result.
+    // NOTE: There are some cases in the spec where we have to compare floats
+    #[allow(clippy::float_cmp)]
     pub fn pow(&self, other: &Self, context: &mut Context<'_>) -> JsResult<Self> {
         Ok(match (self, other) {
             // Fast path:
             (Self::Integer(x), Self::Integer(y)) => u32::try_from(*y)
                 .ok()
                 .and_then(|y| x.checked_pow(y))
                 .map_or_else(|| Self::new(f64::from(*x).powi(*y)), Self::new),
-            (Self::Rational(x), Self::Rational(y)) => Self::new(x.powf(*y)),
-            (Self::Integer(x), Self::Rational(y)) => Self::new(f64::from(*x).powf(*y)),
+            (Self::Rational(x), Self::Rational(y)) => {
+                if x.abs() == 1.0 && y.is_infinite() {
+                    Self::nan()
+                } else {
+                    Self::new(x.powf(*y))
+                }
+            }
+            (Self::Integer(x), Self::Rational(y)) => {
+                if x.abs() == 1 && y.is_infinite() {
```

Comment:
This will overflow if x is `i32::MIN`. It should probably use `checked_abs` instead.

Reply:
I don't think it matters if it overflows since `i32::MIN.abs()` equals `i32::MIN` and it wont satisfy the condition, `checked_abs` might add a jump a condition.

But there is still a problem in debug mode `.abs()` panics on overflow so I should  have used `.wrapping_abs()`.